contracts/StakeManager.sol

```diff
@@ -108,10 +118,20 @@ contract StakeManager is Ownable {
      */
     modifier finalizeEpoch() {
         if (block.timestamp >= epochEnd() && address(migration) == address(0)) {
+            //mp estimation
+            if(expiredMPPerEpoch[currentEpoch] > 0){
+                totalMPPerEpoch -= expiredMPPerEpoch[currentEpoch];
+                delete expiredMPPerEpoch[currentEpoch];
+            }
```

Comment:
I might be missing something here, but isn't that something we can already do when a user `stake()`s?
At the time of staking we're calculating the amount of MPs that have been missed. Those just need to be subtracted from `totalMPPerEpoch`.

Maybe I'm missing something...

Reply:
totalMPPerEpoch would reduce in future, when MAX_BOOST is reached